### PR TITLE
Voice stealing tweaks

### PR DESCRIPTION
**Pull request by paulfd**
Some tweaks on the voice stealing. Comparisons between voices were a bit hazardous in particular, but nothing major.

### CODE REVIEW

File: src/sfizz/Synth.cpp [hunk 1]
```diff
@@ -676,6 +676,9 @@ void sfz::Synth::setSampleRate(float sampleRate) noexcept
 void sfz::Synth::renderVoiceToOutputs(Voice& voice, AudioSpan<float>& tempSpan) noexcept
 {
     const Region* region = voice.getRegion();
+    if (region == nullptr)
```
Comment (jpcima) [inline src/sfizz/Synth.cpp]: How does it happen that a running voice is not associated with a region?

**Comment by jpcima** [comment]
@paulfd, missed entirely to check this PR, sorry about that.
feel free to be insistent next time I forget.

File: src/sfizz/Synth.cpp (hunk 1 above)
Comment (paulfd) [reply src/sfizz/Synth.cpp]: Could be an assert, I had trouble when working on this at some point where voice that were already "killed" were trying to be rendered, it's not necessary anymore.

File: src/sfizz/Voice.h [hunk 2]
```diff
@@ -494,28 +494,31 @@ class Voice {
...
+
+    return true;
 }
 
 inline bool voiceOrdering(const Voice* lhs, const Voice* rhs)
 {
     if (lhs->getAge() > rhs->getAge())
         return true;
-    if (lhs->getAge() < rhs->getAge())
-        return false;
```
Comment (jpcima) [inline src/sfizz/Voice.h:516]: If removing the second check, this makes it incorrect as ordering function. Same of other cases below.

One may also write it:
```
if (a.age != b.age) {
  return a.age > b.age;
}
[...]
return false;
```

**Review by jpcima — APPROVED** [review]

**Pull request merged**